### ISSUE TITLE
test: add fuzz tests for linter.LintContent

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#304]
## Overview

Add Go fuzz tests targeting `linter.LintContent` to close OpenSSF Scorecard alert #46 (Fuzzing). The linter's core parse-and-check path is the highest-value target: arbitrary markdown input is fed through every rule, making it an ideal surface for fuzzing.

## Problem

The Scorecard `Fuzzing` check requires at least one fuzzing integration. Currently the project has no `FuzzXxx` functions, so the score is 0.

### File to create

- `internal/linter/fuzz_test.go`

### Detection logic

```go
func FuzzLintContent(f *testing.F) {
    // seed corpus — valid markdown samples
    f.Add("# Hello\n\nWorld\n")
    f.Add("```go\nfmt.Println()\n```\n")
    // ... additional seeds

    f.Fuzz(func(t *testing.T, data string) {
        linter, err := linter.New(config.Config{})
        if err != nil {
            t.Skip()
        }
        // Must not panic
        linter.LintContent("fuzz.md", data)
    })
}
```

The fuzz function must **not panic** for any input. Violations and errors returned by `LintContent` are expected and should not cause the fuzz function to fail.

### Notes

- Scorecard detects `FuzzXxx` functions via OSS-Fuzz integration or native `go test -fuzz` — simply having a `FuzzXxx` function in the repo is sufficient to satisfy the check.
- Per `CLAUDE.md`, do **not** add a rule-level `_bench_test.go` under `internal/rule/`. The fuzz test belongs in `internal/linter/`.
- Seed corpus should cover: headings, code blocks, lists, links, bare URLs, emphasis, long lines, hard tabs, and empty input.

## Checklist

- [ ] Create `internal/linter/fuzz_test.go` with `FuzzLintContent`
- [ ] Add a representative seed corpus (≥ 8 seeds)
- [ ] Verify `make test` still passes (fuzz functions compile but don't run under normal `go test`)
- [ ] Verify `go test -fuzz=FuzzLintContent -fuzztime=30s ./internal/linter/` finds no panics locally
- [ ] Confirm Scorecard alert #46 closes after the next scheduled scan
